FindOpenSplice.cmake

```diff
@@ -101,12 +115,19 @@ if(NOT DEFINED OSPL_HOME)
 endif(NOT DEFINED OSPL_HOME)
 
 if(WIN32)
-    if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    get_property(csharp_enabled GLOBAL PROPERTY ENABLED_LANGUAGES)
+    if(csharp_enabled MATCHES "CSharp")
+        set(csharp_enabled TRUE)
+    else()
+      set(csharp_enabled FALSE)
+    endif()
+    if(CMAKE_SIZEOF_VOID_P EQUAL 8 OR (csharp_enabled AND CMAKE_CSharp_COMPILER_ARCHITECTURE_ID STREQUAL "x64"))
```

Comment:
Does there exist a corner case where `CMAKE_SIZEOF_VOID_P` is unequal to 8 while `CMAKE_CSharp_COMPILER_ARCHITECTURE` is still `x64`? (Disclaimer: I don't know how C# code gets integrated in a C++ program, so the question may be stupid.)

Reply:
Not sure, but I would expect that at least the normal case would be that the architecture is the same within a single cmake project. I might be wrong, but i think we can handle that if it becomes a problem.